tools/get_dicom_files.pl

```diff
@@ -0,0 +1,292 @@
+subdirectory with name
+
+<pscid>/<visit_label>/<acquisition_date>/<protocol>_<minc_index>
+
+where <minc_index> is the index number of the MINC file to which the DICOMs are associated: e.g. for
+file 'loris_300001_V4_DtiSA_002.mnc', the MINC index is 2 (i.e. the second MINC file with scan type
+'DtiSA'). Finally, a '.tar.gz' that contains all the DICOM files that were extracted is created.
+
+HELP
+my $Usage = <<USAGE;
+#~ perl get_dicom_files.pl [-n patient_name_patterns] [-t scan_type_patterns] [-d tmp_dir] [-o tarBasename] -p profile
+USAGE
+
+
+&Getopt::Tabular::SetHelp( $Help, $Usage );
+&Getopt::Tabular::GetOptions( \@opt_table, \@ARGV )
+    || exit $NeuroDB::ExitCodes::GETOPT_FAILURE;
+$tmpExtractBaseDir = $tmpExtractBaseDir // $ENV{'TMPDIR'} // undef;
+die "You must use either the -n or -t option, or both. Aborting.\n" 
+    if !defined $patientNames && !defined $scanTypes;
+if(!defined $tmpExtractBaseDir or $tmpExtractBaseDir !~ /\S/) {
+    die "The '-d' option was not used and the environment variable TMPDIR is not defined. Aborting.\n";
+}
+ 
+#---------------------------------------#
+# Read prod file to get DB credentials  #
+#---------------------------------------#
+if ( !$profile ) {
+    print $Help;
+    print STDERR "$Usage\n\tERROR: missing -p argument\n\n";
+    exit $NeuroDB::ExitCodes::PROFILE_FAILURE;
+}
+
+{ package Settings; do "$ENV{LORIS_CONFIG}/.loris_mri/$profile" }
+if(!@Settings::db) {
+    die "No database settings in file $ENV{LORIS_CONFIG}/.loris_mri/$profile\n";
+}
+
+my @patientNames       = defined $patientNames ? split(',', $patientNames) : ();
+my @scanTypes          = defined $scanTypes    ? split(',', $scanTypes)    : ();
+my $dbh                = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+my $tarchiveLibraryDir = NeuroDB::DBI::getConfigSetting(\$dbh, 'tarchiveLibraryDir');
+
+#----------------------------------------------------------#
+# Create the directory where DICOM files will be extracted #
+#----------------------------------------------------------#
+my $tmpExtractDir = sprintf("%s/%s.%d", $tmpExtractBaseDir, basename($0), $$);
+mkdir $tmpExtractDir or die "Cannot create directory $tmpExtractDir: $!\n";
+
+#------------------------------------------------------------#
+# Get the list of tarchives for which the patient names      #
+# match the list of supplied patterns (or all tarchives if   #
+# no patterns were specified)                                #
+#------------------------------------------------------------#
+my $query = "SELECT c.PSCID, s.Visit_label, t.DateAcquired, t.ArchiveLocation, t.TarchiveID, t.PatientName "
+          . "FROM tarchive t "
+          . "JOIN session s ON (t.SessionID=s.ID) "
+          . "JOIN candidate c ON (c.CandID=s.CandID) ";
+if(@patientNames) {
+    my @where = map { "t.PatientName LIKE ?" } @patientNames;
+    $query .= sprintf("WHERE %s", join("OR", @where));
```

Comment:
Missing spaces in the where part of the query. Here is what the where part look like (makes the script crash):
`WHERE t.PatientName LIKE ?ORt.PatientName LIKE ?ORt.PatientName LIKE ?`

Reply:
Arghhh! Fixed.